CHANGELOG.rst

```diff
@@ -13,7 +13,7 @@ Added
 
 Changed
 =======
-- Update ``tracepath`` to support two new trace types: ``loop`` and ``incomplete``. Both represent a failure, and ``incomplete`` type also replaces the empty list that was returned as a response in some cases. Other three types are ``starting`` for the first trace_step, ``intermediary`` for subsequent trace_steps (previously ``trace``), and ``last`` for the last trace_step representing successful trace.
+- Update ``tracepath`` to support the trace type: ``loop``. Other three types are ``starting`` for the first trace_step, ``intermediary`` for subsequent trace_steps (previously ``trace``), and ``last`` for the last trace_step representing successful trace.
```

Comment:
> ``last`` for the last trace_step representing successful trace.

This still using the outdated information. We need to better describe `last`. 

Either:

1 - `last` means a terminating last switch where a flow lookup matched. **OR**
2 - `last` means literally just the `last` switch (when there's more than one switch)

Currently we have 2), but then when there's just a single switch, then a client couldn't try to rely on it since it'd be `starting`. It think 1) is more coherent and it has also the potential to be relied on in the future, which means if it isn't `last` then something else happened (a loop or anything else). Ultimately, as Italo said, the client itself has the full context to confirm a success, but `sdntrace_cp` on its own can still also provide a more handy `last` here if we were to go with option 1.


```
{
  "result": [
    {
      "dpid": "00:00:00:00:00:00:00:03",
      "port": 2,
      "time": "2023-07-27 10:41:57.986684",
      "type": "starting",
      "out": {
        "port": 3
      }
    }
  ]
}
```

Can we go for 1)? and then also document it properly, including also documenting `out`? I'd expect that this wouldn't impact on any client.

Reply:
Thank you, @viniarck.

I agree with 1. Anyway, I think that traces with one step have the type last. You can [see this example](https://github.com/kytos-ng/sdntrace_cp/blob/e06700acd587d0d7fa3d73e0b82b289b7ff9f37b/tests/unit/test_main.py#L296-L340) where I made sure of this.
Please, let me know if I am wrong.

Reply:
@gretelliz right, let's go for it then. Appreciate your effort to also try to make it better. 

This test case looks good, but still, for instance, I'm seeing an intra switch EPL still resulting in `starting`, for example (using `linear,4` topology):

```
❯ curl -H 'Content-type: application/json' -X POST http://127.0.0.1:8181/api/kytos/mef_eline/v2/evc/ -d '{"name": "pw_s3", "dynamic_backup_path": true, "uni_a": {"interface_id": "00:00:00:00:00:00:00:03:2"}, "uni_z": { "interface_id": "00:00:00:00:00:00:00:03:3"}}'
{"circuit_id":"d31e80788cb748"}
```

```
❯ curl -H 'Content-type: application/json' -X PUT http://127.0.0.1:8181/api/amlight/sdntrace_cp/v1/trace -d '{"trace": {"switch": {"dpid": "00:00:00:00:00:00:00:03", "in_port": 2}}}' | jq
{
  "result": [
    {
      "dpid": "00:00:00:00:00:00:00:03",
      "port": 2,
      "time": "2023-07-27 15:03:07.582621",
      "type": "starting",
      "out": {
        "port": 3
      }
    }
  ]
}
```

Can you check this example out? I think it's a just a matter of trying to update `starting` as `last` when it has an out port and it's a single result.